### ISSUE TITLE
Central Hub

### ISSUE THREAD

**Issue by DanStaal** [DanStaal/DStaal_UKS_KPBS_Addons#24]
Config update needed.

Habitation/Recycling Info:

| Part                     | PartName                                   | Mass    | Volume   | Seats | Months | Multiplier | EC/s                  | Recycler % | Spreadsheet Approved? |
|:-------------------------|:-------------------------------------------|---------|----------|-------|--------|------------|-----------------------|------------|-----------------------|
| Central Hub              | KKAOSS\_Central\_Hub                       | 7.5     | 30       | 6     | 1.5    | 1.5/5      | 2.25/19.25 (0.0085 W) | 92.5/1     | yes                   |

**Comment by DanStaal** [comment]
Needs a bit of review - configs are partially updated.

Usually Months and Multiplier are separate converters - what's the EC usage of each?

**Comment by Merkov** [comment]
In this case, the hab months are part of the balancing needed for the hab multiplier. Check out the Hab Common module on the MKS Ranger Hab Module for an example. Along with a large multiplier, it adds a few months as well as part of the same converter.

**Comment by DanStaal** [comment]
Ok, so does this look right for it to you?

https://github.com/DanStaal/KPBStoMKS/blob/feature/Life-Support/USI_LS/KPBS_MM_USI_LS.cfg

**Comment by Merkov** [comment]
The numbers aren't quite right. Although I would love to have those stats, the balance spreadsheet shows that the crew capacity should be 5 (this is the hab multiplier's kerbal limit) and the multiplier itself should be 1.5, which is significantly less powerful than what you have listed. I'd love to have a multiplier that high, but the mass just isn't there. 

Also, do we want to keep the resource storage on the hub? I know in the thread we had discussed it not having any storage of its own. Further, since RoverDude said he doesn't expect wear to make a return, do we want to keep all of the patches adding ReplacementParts to vessels? The USI-LS life support config still adds them to all parts with seats, but his parts config files don't actually specify amounts for any parts anymore, including that small radial recycler (which, if I'm not mistaken, means that if he ever did re-activate Wear, that part would not have replacement parts at all).

**Comment by DanStaal** [comment]
Ok, I was just copying off the table above - which is a direct copy of the line in your table in the integration notes.  ;)  Thus why I didn't mark this one closed: I had a feeling there was more we wanted to look at, but I couldn't remember all of it.  I'll drop the resource storage.

I'll have to take a look at the USI configs before I'm sure on ReplacementParts - but my feeling is that we can remove it.

Hmm.  If we drop the resource storage and replace it with Machinery, would that mass be enough to up the crew capacity?  (To me, this part means everyone gets to pull all their work documents and stuff out of their bunks, which means they can separate their work from their off-time, which is a major quality of life improvement.  Just trying to model that in.)

**Comment by Merkov** [comment]
We could add Machinery, but I'm not sure if we want to add an extra resource for only one or two parts. If we decide that a lot of parts could use extra mass, then including the Central Hub in that group would be no issue. I'd hate to "add" an extra resource to those running just USI-LS but not MKS, plus a production chain (even if that chain is just Ore -> Machinery) just to bump the mass of a few parts. Also, if we do decide that being on flat land should equate to a habitation bonus (as is being discussed on the forum thread) then maybe we can just add the extra kerbal and call it fair.

**Comment by DanStaal** [comment]
I'm not against having a third option: Have the parts only roughly balanced with USI-LS, and have a patch in KPBStoMKS that fine-tunes the balance for MKS.  Adding Machinery would be an obvious fine-tune.

Though adding the extra Kerbal sounds cleaner in this case, probably.

**Comment by Merkov** [comment]
Changed the Central Hub config to remove the 1.5 bonus months and add the extra (6th) kerbal space to the multiplier. Also made the multiplier a 1.5 x, not a 5 x. The spreadsheet wants the part to be an extra 1.5 tonnes heavier, but I'd say we're close enough.

**Comment by DanStaal** [comment]
I'm adding 400 Machinery for the MKS-patch version, that balances the mass.  With just USI-LS, you get without, but the balance can be a bit looser then.

I'm not sure about usage rates: I put it in at 0.005, which will deplete the store in 6 months.

**Comment by Merkov** [comment]
Just looking at the spreadsheet, it looks like 300 Machinery would actually bring the dry mass down to 7.45 t, which is pretty much bang-on. 

I'm having second thoughts (sort of) about making the consumption rate really high. My main concern is that machinery levels have a pretty big effect on the rate of conversions (side note here: we don't actually have any conversions occurring on the Central Hub right now... I imagine it's possible to make the purifier and hab multiplier cost machinery and tie their efficiency to machinery, but in MKS they don't. At least, the spreadsheet doesn't think they do). If your parts become half as efficient in a matter of months, that could cause some real headaches. 

For comparison, with 300 Machinery, the spreadsheet says Machinery usage should be 0.000006. I'm not saying it has to be that low, but I'm just not sure anymore.

**Comment by DanStaal** [comment]
I'll admit I mostly just threw it in at the moment - really, this part should be thought of in whole in the end.

I'm tempted to leave out the usage entirely - make that a Stage 3 issue.  I just wanted some reason for it to be there and not have players dump it overboard.  ;)  (And I want to see how that MM patch works in real-life.)  Really, this part is likely going to have to be completely redone in a stage 3 patch.

I'll probably play-test this patch to test MM, and then remove the usage.

**Comment by Merkov** [comment]
Yeah, I think you're right. I think with stage 3, just about every part we've touched here (plus all of the MKS converters we'll need) will use Machinery, but adding Machinery to just one part, then needing to support a resource chain to GET Machinery without MKS, seems a bit silly. It does mean that in base USI-LS games, the Central Hub will be more powerful than it perhaps should be, but I think it's about as close as we'll get. 

Also, keep in mind that with the numbers above, I did NOT include the Admin Module value for the tiny MPL functionality that the Central Hub has. I'm still tempted to throw in a commented-out MM patch that removes MPL functionality from the Central Hub that players (I) can enable if they (I) want.

**Comment by DanStaal** [comment]
I only have the Machinery in the MKS-specific patch, so they would have a resource chain to get it.  I'm hoping most of these parts won't get touched to much in stage 3 - if there's anything that needs to be done on the MKS side for that it should be in the `Part_Mods/LS_Parts.cfg` file, so that we're sure we don't have to change things to much later.  (I don't want to have parts change on people from release to release, if we can avoid it.  Each release touches new parts, and *those* change, but parts that we already dealt with stay dealt with until something upstream changes.)

**Comment by DanStaal** [comment]
Ok, as I suspected trying to add input and output resources to an existing module via MM didn't work very well.  (It probably can be done with more involved syntax, but my quick version didn't work.)  Removing use of Machinery.
